WooCommerce/UITestsFoundation/Screens/Payments/PaymentsScreen.swift

```diff
@@ -41,6 +61,34 @@ public final class PaymentsScreen: ScreenObject {
         return self
     }
 
+    public func tapCollectPayment() throws -> Self {
+        collectPaymentButton.tap()
+        return self
+    }
+
+    public func enterPaymentAmount() throws -> Self {
+        app.enterText(text: "5")
+        nextButton.tap()
+        return self
+    }
+
+    public func takeCashPayment() throws -> Self {
+        takePaymentButton.waitForIsHittable(timeout: 10)
+        takePaymentButton.tap()
+
+        cashPaymentButton.waitForIsHittable(timeout: 10)
+        cashPaymentButton.tap()
+
+        markAsPaidButton.tap()
+        return self
+    }
+
+    @discardableResult
+    public func verifyOrderCompletedToastDisplayed() throws -> Self {
+        XCTAssertTrue(app.staticTexts.matching(NSPredicate(format: "label CONTAINS 'Order completed'")).firstMatch.exists)
```

Comment:
For easier parsing and also debugging, I'd split it the following way:

```suggestion
        let toastElement = app.staticTexts.matching(NSPredicate(format: "label CONTAINS 'Order completed'")).element
        XCTAssertTrue(toastElement.exists)
```

(this is working code).

You can also notice that I replaced `firstMatch` with `element`, which will make test fail in case if we see more than one toast (which should not happen)

Reply:
i've split this so that the element and assert are separated, however, i've decided to keep `firstMatch` instead of `element`. in this case, we are expecting only 1 match, if we use `element`, even when there are multiple matches, we would need to add an additional check to ensure that it returned one instead of multiple (i.e. to check that the array returned 1 value) 

i validated this by updating the predicate to the text "Reader" (appears twice on the screen) and without the additional check, it still passes

while i understand your intention in the comment and agree that it could be good to have the check in case there are multiple toasts, i think in this case, it is more of a nice to have that would slow down the test ([considering that `firstMatch` is faster than `element`](https://developer.apple.com/documentation/xctest/xcuielementquery/1500515-element#discussion))

Reply:
> in this case, we are expecting only 1 match, if we use `element`, even when there are multiple matches, we would need to add an additional check to ensure that it returned one instead of multiple (i.e. to check that the array returned 1 value)

This was a bit of TIL moment for me, since according to [`element` documentation](https://developer.apple.com/documentation/xctest/xcuielementquery/1500515-element), the test should fail when there's more than one match 🤔

I think the fact that nothing is done with the element afterwards (e.g. tap) caused the absence of `Multiple matching elements` automatic fail, when you tried this with `Reader` label. For me, it also did not fail locally, but failed as soon as I tried to tap it (when testing with `Reader`), which makes sense.

> we would need to add an additional check to ensure that it returned one instead of multiple (i.e. to check that the array returned 1 value)

FWIW, we can alternatively achieve this with (notice no usage of `element` or `firstMatch`):

```swift
let orderCompletedToast = app.staticTexts.matching(NSPredicate(format: "label CONTAINS 'Order completed'"))
XCTAssertEqual(1, orderCompletedToast.count)
```